components/director/internal/domain/tenant/fetcher_on_demand_test.go

```diff
@@ -0,0 +1,78 @@
+package tenant_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/tenant/automock"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/tenant"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFetchOnDemand(t *testing.T) {
+	fetchTenantURL := "https://compass-tenant-fetcher.kyma.local/tenants/v1/fetch"
+	tenantID := "b91b59f7-2563-40b2-aba9-fef726037aa3"
+	testErr := errors.New("error")
+
+	testCases := []struct {
+		Name          string
+		Client        func() *automock.Client
+		ExpectedError error
```

Comment:
`string` instead of `error` can also be used here (`ExpectedErrorMsg`), since we only need the error message of the error